src/test/java/com/amazonaws/encryptionsdk/DefaultCryptoMaterialsManagerTest.java

```diff
@@ -275,15 +343,62 @@ public void decrypt_testSimpleRoundTrip() throws Exception {
         }
     }
 
+    @Test
+    public void decrypt_testSimpleRoundTripWithKeyring() throws Exception {
+        for (CryptoAlgorithm algorithm : CryptoAlgorithm.values()) {
+            EncryptionMaterials encryptMaterials = easyGenMaterialsForKeyring(
+                    builder -> builder.setRequestedAlgorithm(algorithm)
+            );
+
+            DecryptionMaterials decryptMaterials
+                    = new DefaultCryptoMaterialsManager(keyring1).decryptMaterials(decryptReqFromMaterials(encryptMaterials));
+
+            assertArrayEquals(decryptMaterials.getDataKey().getKey().getEncoded(),
+                    encryptMaterials.getCleartextDataKey().getEncoded());
+
+            if (encryptMaterials.getTrailingSignatureKey() == null) {
+                assertNull(decryptMaterials.getTrailingSignatureKey());
+            } else {
+                Signature sig = Signature.getInstance(
+                        TrailingSignatureAlgorithm.forCryptoAlgorithm(algorithm).getHashAndSignAlgorithm()
+                );
+
+                sig.initSign(encryptMaterials.getTrailingSignatureKey());
+
+                byte[] data = "hello world".getBytes(StandardCharsets.UTF_8);
+
+                sig.update(data);
+                byte[] signature = sig.sign();
+
+                sig.initVerify(decryptMaterials.getTrailingSignatureKey());
+
+                sig.update(data);
+                sig.verify(signature);
+            }
+        }
+    }
+
     @Test(expected = CannotUnwrapDataKeyException.class)
-    public void decrypt_onDecryptFailure() throws Exception {
+    public void decrypt_onDecryptFailure() {
         new DefaultCryptoMaterialsManager(mock(MasterKeyProvider.class)).decryptMaterials(
                 decryptReqFromMaterials(easyGenMaterials(ignored -> {}))
         );
     }
 
+    @Test(expected = CannotUnwrapDataKeyException.class)
```

Comment:
I suggest you cut a follow-up PR to use assertThrows everywhere, actually.

Reply:
#157